src/transfer.ts

```diff
@@ -214,113 +212,241 @@ async function htmlTransfer() {
+    <form class="transfer-form">
+      <input type="button" value="continue" />
+      <input type="button" value="return" />
+    </form>
+  `;
+  const availableTokenIDs = (<Tokens>account.values.values.get(tokenAddress)).value;
+  const div_checkboxesIDs = document.querySelector<HTMLDivElement>("#checkboxesIDs")!;
+  const checkboxes_IDs = typeof checkedTokenIDs == 'undefined' ? 
+    makeTokenIDsCheckboxes(div_checkboxesIDs, availableTokenIDs) :
+    makeTokenIDsCheckboxes(div_checkboxesIDs, availableTokenIDs, checkedTokenIDs)
+
+  const btn_continue = 
+    document.querySelector<HTMLInputElement>( '.transfer-form input[value="continue"]')!;
+  btn_continue.addEventListener("click", () => advancedTransferContinueButtonEvent(tokenAddress, amount, recipientAddress, checkboxes_IDs!))
+  
+  const btn_return = 
+    document.querySelector<HTMLInputElement>( '.transfer-form input[value="return"]')!;
+  btn_return.addEventListener("click", () => htmlTransfer(tokenAddress, amount, recipientAddress, true))
 }
 
+function makeTokenIDsCheckboxes(div_checkboxesIDs: HTMLDivElement, availableTokenIDs:bigint[], checkedTokenIDs?:bigint[]) : HTMLInputElement[]{
+  const checkboxes_IDs: HTMLInputElement[] = []
+  for (let i = 0; i < availableTokenIDs.length; i++) {
+    const tokenID = availableTokenIDs[i]
+    const checkbox = document.createElement("input")
+    checkbox.type = "checkbox"
+    checkbox.value = `${tokenID}`
+    checkbox.id = `${tokenID}`
+    checkbox.checked = typeof checkedTokenIDs != 'undefined' && checkedTokenIDs.includes(tokenID) ? true : false
+    checkboxes_IDs.push(checkbox)
+    const span =document.createElement("span");
+    span.innerHTML = `${tokenID} </br>`;
+    div_checkboxesIDs.appendChild(checkbox)
+    div_checkboxesIDs.appendChild(span)
+  }
+
+  return checkboxes_IDs
+}
+
+
+
+function advancedTransferContinueButtonEvent(tokenAddress: string, amount: number, recipientAddress: string, checkboxes_IDs: HTMLInputElement[]){
+  const checkboxes_checkedIDs = checkboxes_IDs.filter(checkbox => checkbox.checked)
+  if (checkboxes_checkedIDs.length != amount){
+    alert(`Please choose ${amount} token ID(s)! (currently ${checkboxes_checkedIDs.length} chosen)`)
+    return
+  }
+  const tokenIDs = checkboxes_checkedIDs.map(checkbox=>BigInt(checkbox.value))
+  htmlTransferConfirmation(tokenAddress, amount, recipientAddress, tokenIDs, checkboxes_IDs)
+}
+
+function htmlTransferConfirmation(tokenAddress: string, amount: number, recipientAddress: string, tokenIDs:bigint[], checkboxes_IDs?: HTMLInputElement[]) {
+  div_transfer.innerHTML = `
+    <h2>Please confirm the transfer</h2>
+    <h2>token address: ${tokenAddress}</h2>
+    <h2>amount: ${amount}</h2>
+    <h2>recipient address: ${recipientAddress}</h2>
+    <h2>token ids:<h2>
+    <div id= tokenIDs>
+    </div>
+    <form class="transfer-form">
+      <input type="button" value="make transfer" />
+      <input type="button" value="return" />
+    </form>
+
```

Comment:
html sollte mehr strukturiert sein mit angemesseneren tags, sodass es auch für css einfacher ist den gewünschten Teil zu selektieren und Design aufzusetzen.

Reply:
Wird in design branch umgesetzt.